[CRO-414] Update for use with `tango-and-cash`

jritterbush opened the pull request:
Update the script to be used with our Maxymiser test building library.

- Throttle API calls to avoid rate limiting from MM
- Add additional allowed properties to `elements` methods
- Add create method to `campaign-actions`
- Change out defaults endpoint from 🇪🇺  to 🇺🇸 , because we are of the class of lazy Americans who don't want to override default settings. We want to be the default.
- Use `yarn` to lock down versions better

jritterbush committed:
yarn it up

jritterbush committed:
:us: American Invasion™

:tumbler_glass:

jritterbush committed:
Add handlers for url object and elementId string to element properties

jritterbush committed:
Add create action method

jritterbush committed:
Add throttle to requests

- Add utility to throttle requests to the MM REST API
- Create 2 instances of it one for `get` and one for `put` and `post`
requests

jritterbush committed:
Clarify error message a bit

jritterbush commented:
@ericcecchi @ericmikkelsen @conwaydev and this is also a WIP, but much closer to final from what I can tell. I've modified the original a bit to make it align better with our needs and with the latest options the REST API offers.

ericcecchi commented on __tests__/campaign-actions-test.js:
```diff
@@ -4,8 +4,8 @@ const request = require('superagent');
-var basePath = 'https://api-eu.maxymiser.com/v1';
+var authPath = 'https://api-auth-us.maxymiser.com/oauth2/v1';
+var basePath = 'https://api-us.maxymiser.com/v1';
 var credentials = {
```
You could add a test here for the new `create` method.

jritterbush replied:
done and passing

ericcecchi commented on README.md:
```diff
@@ -409,6 +412,42 @@ api.campaigns.actions.get({
+  elementName: 'renderer',
+  // siteId: 'MzIxMzM',
+  // campaignId: 'MDA2MjYx',
+  // elementId: 'MDMyMDU4',
```
`elementId` and `elementName` aren't actually part of the actions request schema.

jritterbush committed:
Create test for new actions.create; Fix test issues

- Add test for campaign actions create and added mock response
- For clarity changed actions test paths to be campaign/actions to match
campaign/scripts (vs site/actions and site/scripts)
- Fixed broken urls in auth tests :us:

jritterbush committed:
Remove element refs from actions

jritterbush committed:
Remove action options and add missing script options
